apollo-router/src/query_planner/fetch.rs

```diff
@@ -266,10 +266,30 @@ impl FetchNode {
             // when errors have been redacted in the include_subgraph_errors module.
             // Unfortunately, not easy to fix here, because at this point we don't
             // know if we should be redacting errors for this subgraph...
-            .map_err(|e| FetchError::SubrequestHttpError {
-                status_code: None,
-                service: service_name.to_string(),
-                reason: e.to_string(),
+            .map_err(|e| {
+                if !e.is::<FetchError>() {
+                    FetchError::SubrequestHttpError {
+                        status_code: None,
+                        service: service_name.to_string(),
+                        reason: e.to_string(),
+                    }
+                } else {
+                    let inner = e.downcast::<FetchError>().unwrap();
```

Comment:
You should not call `.unwrap()` but just match over the `Option` instead.

Reply:
Good catch! That is correct.